volumina/pixelpipeline/imagesources.py

```diff
@@ -420,8 +423,95 @@ def toImage( self ):
                 #FIXME: maybe this should be done in a better way using an operator before the colortable request which properly handles 
                 #this problem 
                 warnings.warn("Data for colortable layers cannot be float, casting",RuntimeWarning)
-                a=a.astype(np.int32)
-            vigra.colors.applyColortable(a, self._colorTable, byte_view(img))
+                a = np.asanyarray(a, dtype=np.uint32)
+
+            # If we have a masked array with a non-trivial mask, ensure that mask is made transparent.
+            _colorTable = self._colorTable
+            if np.ma.is_masked(a):
```

Comment:
On a side note, do you know what the max allowable type for the label image is for `vigra.colors.applyColortable`. It seems to be related to the function `pythonApplyColortable` ( https://github.com/ukoethe/vigra/blob/4510b643cbf1d62d1b9111c53236919eb1c2d063/vigranumpy/src/core/colors.cxx#L302 ), but the type isn't obvious. I have assumed uint32, which seem to work, but if it is uint64 I wouldn't want to unnecessarily disallow something.